src/main/java/backend/resource/TurboLabel.java

```diff
@@ -211,4 +223,130 @@ public int hashCode() {
+                .stream()
+                .filter(name -> {
+                    if (hasGroup(name)) {
+                        return Utility.containsIgnoreCase(getGroup(name), labelGroup);
+                    }
+                    return false;
+                })
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the first label that matches the keyword
+     * i.e. the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static String getMatchedLabelName(List<String> repoLabels, String keyword) {
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByPartialName(newMatchedLabels, getName(keyword));
+        newMatchedLabels = filterByPartialGroupName(newMatchedLabels, getGroup(keyword));
+        return newMatchedLabels.get(0);
+    }
+
+    /**
+     * Returns true if there is exactly 1 matching label for keyword
+     *
+     * A label is matching if:
+     * the label's group contains keyword's group and label's name contains keyword's name
+     * @param repoLabels
+     * @param keyword
+     * @return
+     */
+    public static boolean hasExactlyOneMatchedLabel(List<String> repoLabels, String keyword) {
+        List<String> newMatchedLabels = new ArrayList<>();
+        newMatchedLabels.addAll(repoLabels);
+        newMatchedLabels = filterByPartialName(newMatchedLabels, getName(keyword));
+        newMatchedLabels = filterByPartialGroupName(newMatchedLabels, getGroup(keyword));
+        return newMatchedLabels.size() == 1;
+    }
+
+    /**
+     * Identifies if a label belongs to an exclusive group. Only one label
+     * from an exclusive group is allowed to be assigned to an issue
+     * @param labelName
+     * @return
+     */
+    public static boolean hasExclusiveGroup(String labelName) {
+        return TurboLabel.getDelimiter(labelName).isPresent() && TurboLabel.getDelimiter(labelName).get().equals(".");
+    }
+
+    /**
+     * Determines the group that labelName belongs to
+     * A labelName is considered to be in a group if getDelimiter(labelName).isPresent() is true.
+     * @param labelName
+     * @return
+     */
+    public static String getGroup(String labelName) {
+        if (!hasGroup(labelName)) return "";
+        return labelName.substring(0, labelName.indexOf(TurboLabel.getDelimiter(labelName).get()));
+    }
+
+    /**
+     * Returns the name of labelName after removing its group
+     * @param labelName
+     * @return
+     */
+    public static String getName(String labelName) {
+        if (!hasGroup(labelName)) return labelName;
+        return labelName.substring(labelName.indexOf(TurboLabel.getDelimiter(labelName).get()) + 1);
+    }
+
+    public static boolean hasGroup(String name) {
+        return TurboLabel.getDelimiter(name).isPresent();
+    }
```

Comment:
These methods that take a label name as the only parameter sound more suitable as instance methods.
e.g. `(new TurboLabel(labelName)).hasGroup()`